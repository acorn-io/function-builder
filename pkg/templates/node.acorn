let baseImage: "node:18"
let runtimeImage: "node:18"

localFiles: {
	".dockerignore": """
node_modules/

"""
	"package.json": ```
{
  "name": "function",
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
  },
  "dependencies": {
      "express": "^4.18.2"
  },
  "private": true
}
    ```
}

detect: [{
	files: ["package.json"]
	priority: 100
}, {
	files: ["index.js"]
	priority: 50
}]

targets: dev: {
	from: baseImage
	acornfile: """
		dirs: {
			"/src": "./\\(build.context)"
		}
		build: watchFiles: [
			"\\(build.context)/package.json",
			"\\(build.context)/package-lock.json",
			"\\(build.context)/env-setup.sh",
		]
	"""

	entrypoint: ["nodemon", "index.js"]
	run: [
		{
			copy: {
				source: "env-setup.sh*"
				dest:   "/src"
			}
		},
		"""
        if [ -e env-setup.sh ]; then
            chmod +x env-setup.sh
            ENV_SETUP_PHASE=dev ./env-setup.sh
        fi
        """,
		"npm install -g nodemon",
		{
			copy: {
				source: "package.json package-lock.json*"
				dest:   "/src"
			}
		},
		"npm install",
		{
			copy: {
				source: "."
				dest:   "/src"
			}
		},
	]
}

targets: default: {
	from: "dev"
	user: 10000
}
