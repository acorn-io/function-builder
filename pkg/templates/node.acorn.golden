{
  "targets": {
    "default": {
      "from": "dev",
      "workdir": "/src",
      "user": 10000
    },
    "dev": {
      "from": "node:18",
      "entrypoint": [
        "nodemon",
        "index.js"
      ],
      "workdir": "/src",
      "acornfile": "\tdirs: {\n\t\t\"/src\": \"./\\(build.context)\"\n\t}\n\tbuild: watchFiles: [\n\t\t\"\\(build.context)/package.json\",\n\t\t\"\\(build.context)/package-lock.json\",\n\t\t\"\\(build.context)/env-setup.sh\",\n\t]",
      "run": [
        {
          "copy": {
            "source": "env-setup.sh*",
            "dest": "/src"
          }
        },
        {
          "command": "if [ -e env-setup.sh ]; then\n    chmod +x env-setup.sh\n    ENV_SETUP_PHASE=dev ./env-setup.sh\nfi"
        },
        {
          "command": "npm install -g nodemon"
        },
        {
          "copy": {
            "source": "package.json package-lock.json*",
            "dest": "/src"
          }
        },
        {
          "command": "npm install"
        },
        {
          "copy": {
            "source": ".",
            "dest": "/src"
          }
        }
      ]
    }
  },
  "localFiles": {
    ".dockerignore": "node_modules/\n",
    "package.json": "\n{\n  \"name\": \"function\",\n  \"version\": \"0.0.1\",\n  \"description\": \"\",\n  \"main\": \"index.js\",\n  \"scripts\": {\n  },\n  \"dependencies\": {\n      \"express\": \"^4.18.2\"\n  },\n  \"private\": true\n}\n    "
  },
  "detect": [
    {
      "priority": 100,
      "files": [
        "package.json"
      ]
    },
    {
      "priority": 50,
      "files": [
        "index.js"
      ]
    }
  ]
}